lib/Conch/Validation/HddSize.pm

```diff
@@ -0,0 +1,53 @@
+package Conch::Validation::HddSize;
+
+use Mojo::Base 'Conch::Validation', -signatures;
+use Mojo::JSON 'from_json';
+
+use constant name        => 'hdd_size';
+use constant version     => 1;
+use constant category    => 'DISK';
+use constant description => 'Check hard drive sizes';
+
+sub validate ($self, $data) {
+    my $hw_spec = from_json($self->hardware_product_specification // {});
+
+    return if not $hw_spec->{disk_size};
+
+    foreach my $disk_serial (keys $data->{disks}->%*) {
+        my $drive_model = $data->{disks}{$disk_serial}{model};
+        if (not $drive_model) {
+            $self->fail('missing drive model for disk '.$disk_serial,
```

Comment:
I think we should just skip the validation if we don't have data to compare, rather than fail

Reply:
Could do. I *think* model is a mandatory field however? (Not that our schema says it is, but my reading of the livesys code suggests that it's always there.)

Reply:
If we can get confirmation of that and get it into the schema or at least documentation/comments, then I'm fine with failing. 

I should also have put a similar comment in the next `->fail` as well.

Reply:
If we have a disk_size in the hardware specification we are expecting to have that data in the report, if it’s not in the report something is wrong.

Reply:
Understood.